### PR TITLE
feat: add linter command

### PR DESCRIPTION
**Pull request by cds-amal**
This PR Introduces a new linter command to validate txtx runbooks and manifests before execution.

# Linter Command

Adds `txtx lint` to validate runbooks and manifests before execution.

## Usage

```bash
# Lint workspace runbooks
txtx lint

# Lint specific runbook
txtx lint my_runbook [--env my-env] [--input name=value]...

# CI integration
txtx lint --format json
```

**Flags:**
- `--gen-cli` / `--gen-cli-full` - Generate CLI usage
- `--manifest-path PATH` - Custom manifest location
- `--env ENV` - Environment selection
- `--input KEY=VALUE` - Input overrides
- `--format [stylish|compact|json|quickfix|github|csv]` - Output format
- `--init` - Generate `.txtxlint.yml` config

## Validation Rules

1. **undefined-input** - Input references without definitions
2. **naming-convention** - Enforces snake_case inputs
3. **cli-override** - CLI inputs overriding manifest values
4. **sensitive-data** - Potential secrets in values

## Configuration

Optional `.txtxlint.yml` for rule customization (experimental, format may change).

## Testing

```bash
just test-lint-unit   # Rule logic
just test-lint-int    # End-to-end
just test-lint-prop   # Edge cases
```

## Implementation

- Uses txtx-core validation infrastructure
- Workspace-aware manifest analysis
- Shares runbook resolution with `txtx run`
- No supervisor UI dependencies

### CODE REVIEW

File: crates/txtx-cli/src/cli/runbooks/mod.rs [hunk 1]
```diff
@@ -727,7 +727,7 @@ pub async fn handle_run_command(
     let block_store = Arc::new(RwLock::new(BTreeMap::new()));
     let log_store = Arc::new(RwLock::new(Vec::new()));
     let (kill_loops_tx, kill_loops_rx) = channel::bounded(1);
-    let (action_item_events_tx, action_item_events_rx) = tokio::sync::broadcast::channel(32);
+    let (_action_item_events_tx, action_item_events_rx) = tokio::sync::broadcast::channel(32);
```
Comment (MicaiahReid) [inline crates/txtx-cli/src/cli/runbooks/mod.rs]: This will break things when the `supervisor_ui` feature is enabled

File: crates/txtx-cli/src/cli/common/addon_registry.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+use txtx_addon_network_bitcoin::BitcoinNetworkAddon;
+use txtx_addon_network_evm::EvmNetworkAddon;
+use txtx_addon_network_svm::SvmNetworkAddon;
+use txtx_addon_telegram::TelegramAddon;
+use txtx_core::kit::Addon;
+use txtx_core::std::StdAddon;
+
+/// Get all available addons as a shared reference
+pub fn get_all_addons() -> Arc<Vec<Box<dyn Addon>>> {
+    let addons: Vec<Box<dyn Addon>> = vec![
```
[the remark this answers is not in the document]
Comment (cds-amal) [reply crates/txtx-cli/src/cli/common/addon_registry.rs]: ooooh. yea. good catch!
